src/JSInterop/Microsoft.JSInterop/src/JSRuntimeBase.cs

```diff
@@ -62,18 +62,31 @@ public Task<T> InvokeAsync<T>(string identifier, params object[] args)
         /// <param name="argsJson">A JSON representation of the arguments.</param>
         protected abstract void BeginInvokeJS(long asyncHandle, string identifier, string argsJson);
 
-        internal void EndInvokeDotNet(string callId, bool success, object resultOrException)
+        /// <summary>
+        /// Allows derived classes to configure the information about an exception in a JS interop call that gets sent to JavaScript.
+        /// </summary>
+        /// <remarks>
+        /// This callback can be used in remote JS interop scenarios to sanitize exceptions that happen on the server to avoid disclosing
+        /// sensitive information to remote browser clients.
+        /// </remarks>
+        /// <param name="exception">The exception that occurred.</param>
+        /// <param name="assemblyName">The assembly for the invoked .NET method.</param>
+        /// <param name="methodIdentifier">The identifier for the invoked .NET method.</param>
+        /// <returns>An object containing information about the exception.</returns>
+        protected virtual object OnDotNetInvocationException(Exception exception, string assemblyName, string methodIdentifier) => exception.ToString();
```

Comment:
I kinda wonder about this signature being fragile, but I don't have any better ideas. I think this is good for right now and we can look at it again as part of the API review of this area.